File: school/summer2013/example_openmp/fortran/19_for_nowait.f90
program nowait_for
    integer, parameter :: N = 9999
    integer i, j, a(0:N, 0:N)

    call omp_set_num_threads(4)
!$omp parallel private(i,j)
    !$omp do
    do j=0, N
        do i=0, j
            a(i,j) = i+j
        end do
    end do
    !$omp end do nowait

    !$omp do
    do j=0, N
        do i=j+1, N
            a(i,j) = i-j
        end do
    end do
!$omp end parallel
end

File: summer2013/example_openmp/c/19_for_nowait.c
#include <stdio.h>
#define N 20000

int a[N][N];

int main()
{
	int i, j;

#pragma omp parallel private(i,j)
{
	#pragma omp for nowait
	for(i=0; i<N; i++)
		for(j=i; j<N; j++)
			a[i][j] = i+j;

	#pragma omp for
	for(i=0; i<N; i++)
		for(j=0; j<i; j++)
			a[i][j] = i-j;
} // pragma omp parallel
}
